catroid/src/org/catrobat/catroid/ui/dialogs/OrientationDialog.java

```diff
@@ -143,4 +122,49 @@ public void setCreateEmptyProject(boolean isChecked) {
 	public void setCreateDroneProject(boolean isChecked) {
 		createDroneProject = isChecked;
 	}
+
+	private class LoadingOnCreatingProject extends AsyncTask<String, Void, Boolean> {
+		private ProgressDialog progressDialog;
+
+		protected LoadingOnCreatingProject() {
+			progressDialog = new ProgressDialog(getActivity());
+		}
+
+		@Override
+		protected void onPostExecute(Boolean showing) {
+			progressDialog.dismiss();
+		}
+
+		@Override
+		protected void onPreExecute() {
+			this.progressDialog.setMessage(getResources().getString(R.string.please_wait));
+			this.progressDialog.show();
+		}
+
+		@Override
+		protected Boolean doInBackground(String... params) {
+
+			try {
+					ProjectManager.getInstance().initializeNewProject(projectName, getActivity(), createEmptyProject, createDroneProject, createLandscapeProject);
+				} catch (IllegalArgumentException illegalArgumentException) {
+					Utils.showErrorDialog(getActivity(), R.string.error_project_exists);
+				} catch (IOException ioException) {
+					Utils.showErrorDialog(getActivity(), R.string.error_new_project);
+					Log.e(TAG, Log.getStackTraceString(ioException));
+					dismiss();
+				}
+
+				Intent intent = new Intent(getActivity(), ProjectActivity.class);
```

Comment:
As `Activity.startActivity(Intent)` runs on the UI Thread anyway, I'd rather put the Intent creation and Activity start into the `onPostExecute()` which also runs o the UI Thread.
The long term task (which `doInBackground` is designed for) here is the `InitializeNewProject()`